contracts/managers/SocialTradingManager.sol

```diff
@@ -0,0 +1,296 @@
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.5.7;
+pragma experimental "ABIEncoderV2";
+
+import { SafeMath } from "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import { ICore } from "set-protocol-contracts/contracts/core/interfaces/ICore.sol";
+import { ILiquidator } from "set-protocol-contracts/contracts/core/interfaces/ILiquidator.sol";
+import { IRebalancingSetTokenV2 } from "set-protocol-contracts/contracts/core/interfaces/IRebalancingSetTokenV2.sol";
+import { ISetToken } from "set-protocol-contracts/contracts/core/interfaces/ISetToken.sol";
+import { RebalancingLibrary } from "set-protocol-contracts/contracts/core/lib/RebalancingLibrary.sol";
+
+import { ISocialAllocator } from "./allocators/ISocialAllocator.sol";
+
+
+/**
+ * @title SocialTradingManager
+ * @author Set Protocol
+ *
+ * Singleton manager contract through which all social trading sets are managed. Traders can choose a percentage
+ * between 0 and 100 for each rebalance. The trading pair used for each trading pool is defined by the allocator
+ * passed in on pool creation. Only compatible with RebalancingSetTokenV2 constracts. All permissioned functions
+ * on the RebalancingSetTokenV2 must be called through the administrative functions exposed on this contract.
+ */
+contract SocialTradingManager {
+    using SafeMath for uint256;
+
+    /* ============ Structs ============ */
+    struct PoolInfo {
+        address trader;
+        ISocialAllocator allocator;
+        uint256 currentAllocation;
+    }
+
+    /* ============ Events ============ */
+
+    event TradingPoolCreated(
+        address indexed _trader,
+        ISocialAllocator indexed _allocator,
+        address _tradingPool,
+        uint256 _startingAllocation
+    );
+
+    event AllocationUpdate(
+        address indexed _tradingPool,
+        uint256 _oldAllocation,
+        uint256 _newAllocation
+    );
+
+    event NewTrader(
+        address indexed _tradingPool,
+        address _oldTrader,
+        address _newTrader
+    );
+
+    /* ============ Modifier ============ */
+
+    modifier onlyTrader(IRebalancingSetTokenV2 _tradingPool) {
+        require(
+            msg.sender == pools[address(_tradingPool)].trader,
+            "Sender must be trader"
+        );
+        _;
+    }
+
+    /* ============ Constants ============ */
+
+    uint256 public constant REBALANCING_SET_NATURAL_UNIT = 10 ** 6;
+    uint256 public constant SCALE_FACTOR = 10 ** 18;
```

Comment:
Let's consider exposing this in `CommonMath` as a function so it doesn't need to be redeclared every file

Reply:
+1